File: library/src/main/java/com/gmail/fuskerr63/android/library/birthday/NotifyNotificationManagerImpl.java
package com.gmail.fuskerr63.android.library.birthday;

import com.gmail.fuskerr63.java.interactor.NotifyNotificationManager;
import com.gmail.fuskerr63.java.interactor.NotificationRepository;

import io.reactivex.annotations.NonNull;
import io.reactivex.annotations.Nullable;

public class NotifyNotificationManagerImpl implements NotifyNotificationManager {
    @NonNull
    private final NotificationRepository notificationRepository;

    private static final String CHANNEL_ID = "CHANNEL_ID";
    private final int flagUpdateCurrent;
    private final int priority;

    public NotifyNotificationManagerImpl(
            @NonNull NotificationRepository notificationRepository,
            int flagUpdateCurrent,
            int priority) {
        this.notificationRepository = notificationRepository;
        this.flagUpdateCurrent = flagUpdateCurrent;
        this.priority = priority;
    }


    @Override
    public void notifyNotification(int id, @Nullable String text) {
        notificationRepository.notifyNotification(id, text, flagUpdateCurrent, CHANNEL_ID, priority);
    }
}

File: library/src/main/java/com/gmail/fuskerr63/android/library/birthday/NotifyNotificationManagerImpl.kt
package com.gmail.fuskerr63.android.library.birthday

import android.app.PendingIntent
import androidx.core.app.NotificationCompat
import com.gmail.fuskerr63.java.interactor.NotificationRepository
import com.gmail.fuskerr63.java.interactor.NotifyNotificationManager

class NotifyNotificationManagerImpl(
    private val notificationRepository: NotificationRepository
) : NotifyNotificationManager {
    companion object {
        const val CHANNEL_ID: String = "CHANNEL_ID"
    }

    override fun notifyNotification(id: Int, text: String) {
        notificationRepository.notifyNotification(
            id,
            text,
            PendingIntent.FLAG_UPDATE_CURRENT,
            CHANNEL_ID,
            NotificationCompat.PRIORITY_DEFAULT
        )
    }
}
